Fixes #16723: Update documentation with missing informations (#237)
Normation/rudder-plugins: user-management/README.adoc

|read_only | Can access to every read only part, can perform no action
|inventory | Access to information about nodes, can see their inventory, but can't act on them
|rule_only | Access to information about rules, but can't modify them

> |workflow  | Access to all workflow usage like validate and deploye changes

Reviewer: deploy

|compliance| Can access and act on compliance section

> |deployer  | Access to workflow usage and compliance.

Reviewer: no dot at the end for consistency

* *configuration_read* -> Will give read access to the configuration (Rule management, Directives and Parameters)
* *rule_write, node_read* -> Will give read and write access to the Rules and read access to the Nodes

.Authorization's rights
image::docs/images/details_rights.png[]

== Password Hash

> The plugin use the hash function set globally to hash and store your password, by default RUDDER used https://en.wikipedia.org/wiki/Bcrypt[BCrypt]. +

Reviewer: ```suggestion
The plugin uses the hash function set globally to hash and store your password (https://en.wikipedia.org/wiki/Bcrypt[BCrypt] by default). +
```

Reviewer: What does "set globally" mean? Where is it set?

> You can both provide a hash or provide a plain text password who will be hashed and then stored, according to the function set globally.

Reviewer: ```suggestion
You can both provide a hash or provide a plain text password which will be hashed and then stored, according to the function set globally.
```

.BCrypt parameters

> [WARNING]

Reviewer: I would make it a `NOTICE` instead

===========

> Since BCrypt is the default hash function, by default RUDDER use **2b** as version and the *cost is set to 12*. If you want to change the cost you have to add `rudder.bcrypt.cost` properties in `/opt/rudder/etc/rudder-web.properties` +

Reviewer: ```suggestion
By default RUDDER uses bcrypt with **2b** as version and the *cost set to 12*. If you want to change the cost you need set it in the `rudder.bcrypt.cost` property in `/opt/rudder/etc/rudder-web.properties` +
```

* *configuration_read* -> Will give read access to the configuration (Rule management, Directives and Parameters)
* *rule_write, node_read* -> Will give read and write access to the Rules and read access to the Nodes

.Authorization's rights
image::docs/images/details_rights.png[]

== Password Hash

The credentials of a user are defined in the XML file
`/opt/rudder/etc/rudder-users.xml`. This file expects the following format:

----

<authentication hash="bcrypt">
  <user name="Jane Doe"  password="xxxxxxx" role="inventory,compliance,rule_only"/>
  <user name="admin"    password="xxxxxxx" role="administrator"/>
  <user name="John Doe" password="xxxxxxx" role="no_rights"/>
</authentication>

----

> The plugin uses the hash function set in this file to hash and store your password, by default RUDDER used https://en.wikipedia.org/wiki/Bcrypt[BCrypt]. +

Reviewer: ```suggestion
The plugin uses the hash function set in this file to hash and store your password (which is https://en.wikipedia.org/wiki/Bcrypt[BCrypt] by default). +
```
